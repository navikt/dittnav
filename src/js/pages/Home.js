import React, { Component } from 'react';

import Vta from 'js/components/VTA';
import PersonInfo from 'js/components/PersonInfo';
import InfoMeldinger from 'js/components/InfoMeldinger';
import Tjenester from 'js/components/Tjenester';
import Lenkelister from 'js/components/Lenkelister';
import Artikkel from 'js/components/Artikkel';
import Unleash from 'js/components/Unleash';
import PropTypes from 'prop-types';
import DelayedSpinner from 'js/components/DelayedSpinner';
import Api from 'js/Api';

const getInfoMeldinger = (info, paabegynteSoknader, mininnboks) => ({
  isInactive: info.personinfo ? info.personinfo.inaktiv : true,
  isMeldeKortUser: info.personinfo ? info.personinfo.meldekortbruker : null,
  infoMessages: info.infoMessages,
  agMessagesCount: info.agMessagesCount,
  navMessagesCount: info.navMessagesCount,
  paabegynteSoknader,
  meldekort: info.meldekort,
  isRegisteredAtIArbeid: info.personinfo ? info.personinfo.erUnderRegistreringIArbeid : null,
  mininnboks,
});

class Home extends Component {
  render() {
    const { info, paabegynteSoknader, mininnboks, fetching } = this.props;
    const C = props => (props.isFeatureEnabled ? <Vta /> : <Tjenester services={info.viktigeTjenester} />);
    const tjeneserEllerVta = info.personinfo && info.personinfo.underOppfolging ? <Unleash api={Api} feature="dittnav.fo"><C /></Unleash> : <Tjenester services={info.viktigeTjenester} />;
    return (
      <React.Fragment>
        <div className="row">
          <div className="maincontent side-innhold">
            <div className="col-md-12">
              <PersonInfo personInfo={info.personinfo} />
              {fetching < 3 ? <DelayedSpinner delay={500} spinnerClass="header-spinner" /> : null}
              <InfoMeldinger {...getInfoMeldinger(info, paabegynteSoknader, mininnboks)} />
              { tjeneserEllerVta }
              <Lenkelister links={info.andreTjenester} />
            </div>
          </div>
        </div>
        <Artikkel article={info.article} />
      </React.Fragment>
    );
  }
}

Home.propTypes = {
  info: PropTypes.any.isRequired, // eslint-disable-line react/forbid-prop-types
  paabegynteSoknader: PropTypes.any, // eslint-disable-line react/forbid-prop-types
  mininnboks: PropTypes.any.isRequired, // eslint-disable-line react/forbid-prop-types
  fetching: PropTypes.number.isRequired,
};

Home.defaultProps = {
  paabegynteSoknader: null,
};

export default Home;
