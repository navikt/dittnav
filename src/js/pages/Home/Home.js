import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Undertittel } from 'nav-frontend-typografi';
import { FormattedMessage as F } from 'react-intl';
import Vta from '../../components/VTA';
import PersonInfo from '../../components/PersonInfo';
import InfoMeldinger from '../../components/InfoMeldinger';
import DittnavFliser from '../../components/DittnavFliser';
import DittnavLenkePanel from '../../components/DittnavLenkePanel';
import Lenkelister from '../../components/Lenkelister';
import DelayedSpinner from '../../components/DelayedSpinner';
import Config from '../../globalConfig';

class Home extends Component {
  render() {
    const { oppfolging, meldekort, person, identifikator, paabegynteSoknader, mininnboks, sakstema, hendelser, updateHendelser, oppfolgingHasLoaded, loading } = this.props;
    const erUnderOppfolging = oppfolging && oppfolging.erBrukerUnderOppfolging;
    const generelleEllerVta = erUnderOppfolging ? <Vta /> : <DittnavFliser />;
    const oppfolgingsLenker = Config.dittNav.OPPFOLGINGS_LENKER;
    const generelleLenker = Config.dittNav.GENERELLE_LENKER;

    return (
      <>
        <div className="row">
          <div className="maincontent side-innhold">
            <div className="col-md-12" id="dittnav-main-container">
              <PersonInfo person={person} identifikator={identifikator} />
              { loading ? <DelayedSpinner delay={500} spinnerClass="header-spinner" /> : null }
              <InfoMeldinger meldekort={meldekort} paabegynteSoknader={paabegynteSoknader} mininnboks={mininnboks} hendelser={hendelser} updateHendelser={updateHendelser} />
              <DittnavLenkePanel sakstema={sakstema} />
              { oppfolgingHasLoaded ? generelleEllerVta : null }
              <Undertittel className="relatert-informasjon__subheader">
                <F id="relatertInformasjon.header" />
              </Undertittel>
              { erUnderOppfolging ? <Lenkelister links={oppfolgingsLenker} /> : <Lenkelister links={generelleLenker} /> }
            </div>
          </div>
        </div>
      </>
    );
  }
}

Home.propTypes = {
  oppfolging: PropTypes.any, // eslint-disable-line react/forbid-prop-types
  meldekort: PropTypes.any, // eslint-disable-line react/forbid-prop-types
  person: PropTypes.any, // eslint-disable-line react/forbid-prop-types
  identifikator: PropTypes.any, // eslint-disable-line react/forbid-prop-types
  paabegynteSoknader: PropTypes.any, // eslint-disable-line react/forbid-prop-types
  mininnboks: PropTypes.any.isRequired, // eslint-disable-line react/forbid-prop-types
  sakstema: PropTypes.any.isRequired, // eslint-disable-line react/forbid-prop-types
  oppfolgingHasLoaded: PropTypes.any.isRequired, // eslint-disable-line react/forbid-prop-types
  hendelser: PropTypes.any, // eslint-disable-line react/forbid-prop-types
  updateHendelser: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
};

Home.defaultProps = {
  oppfolging: null,
  meldekort: null,
  person: null,
  identifikator: null,
  paabegynteSoknader: null,
  hendelser: null,
};

export default Home;
