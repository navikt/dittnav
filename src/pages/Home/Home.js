import React from 'react';
import { FormattedMessage as F } from 'react-intl';
import { Undertittel } from 'nav-frontend-typografi';
import { useIsFetching } from 'react-query';
import { useOppfolging } from '../../hooks/usePerson';
import { generelleLenker, oppfolgingsLenker } from '../../utils/lenker';
import useStore from '../../hooks/useStore';
import PersonInfo from '../../components/PersonInfo';
import InfoMeldinger from '../../components/InfoMeldinger';
import KoronaSpesial from '../../components/korona-spesial/KoronaSpesial';
import DittnavLenkePanel from '../../components/DittnavLenkePanel';
import Lenkelister from '../../components/Lenkelister';
import DelayedSpinner from '../../components/common/DelayedSpinner';
import InnloggingsModal from '../../components/common/InnloggingsModal';
import Brodsmuler from '../../utils/brodsmuler';
import Vta from '../../components/VTA';
import PageBase from '../PageBase';
import DittnavFliser from '../../components/DittnavFliser'; 
import UnleashWrapper from '../../components/UnleashWrapper'

const Home = () => {
  const [{ data: oppfolging, isLoading: oppfolgingIsLoading, isError: oppfolgingIsError }] = useOppfolging();
  const isFetching = useIsFetching();
  const { state } = useStore();

  const erUnderOppfolging = oppfolging && oppfolging.content && oppfolging.content.erBrukerUnderOppfolging;
  const generelleEllerVta = oppfolgingIsError ? <DittnavFliser /> : <Vta />;

  const NyVtaForStandardInnsats = ({isLoading, isError}) => {
    if (isLoading ||Â isError) return null;
    return (
      <UnleashWrapper toggle='situasjon'>
        <h1>Hallo</h1>
        <Vta kreverStandardInnsatsgruppe />
      </UnleashWrapper>
    )
  }
  //const vtaForStandardInnsats = oppfolgingIsError ? null : <NyVtaForStandardInnsats />;

  if (state.visInnloggingsModal) {
    return (<InnloggingsModal onClick={() => null} isOpen />);
  }

  return (
    <PageBase uniqueErrors={state.error} breadcrumbs={Brodsmuler.dittnav}>
      <div className="row">
        <div className="maincontent side-innhold">
          <div className="col-md-12" id="dittnav-main-container">
            <PersonInfo />
            {isFetching ? <DelayedSpinner delay={500} spinnerClass="header-spinner" /> : null}
            <InfoMeldinger />
            <NyVtaForStandardInnsats isLoading={oppfolgingIsLoading} isError={oppfolgingIsError}/>
            <KoronaSpesial />
            <DittnavLenkePanel />
            {oppfolgingIsLoading ? null : generelleEllerVta}
            <Undertittel className="flere-tjenester__subheader">
              <F id="flere.tjenester.header" />
            </Undertittel>
            {erUnderOppfolging
              ? <Lenkelister links={oppfolgingsLenker} />
              : <Lenkelister links={generelleLenker} />}
          </div>
        </div>
      </div>
    </PageBase>
  );
};

export default Home;
