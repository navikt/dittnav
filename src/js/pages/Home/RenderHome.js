import React, { useEffect, useState } from 'react';
import useBeskjedStore from '../../hooks/useBeskjedStore';
import PageFrame from '../PageFrame';
import Home from './Home';
import { ADD_BESKJEDER } from '../../types/Actions';
import ApiType from '../../types/ApiType';

const RenderHome = ({ api }) => {
  const [data, setData] = useState({
    oppfolging: null,
    meldekort: null,
    person: null,
    identifikator: null,
    paabegynteSoknader: null,
    mininnboks: [],
    sakstema: { antallSakstema: 0, sakstemaList: [] },
    innlogging: null,
    oppgaver: null,
    innbokser: null,
    antallBrukernotifikasjoner: 0,
    errors: [],
    fetching: 0,
    oppfolgingHasLoaded: false,
  });

  const { dispatch } = useBeskjedStore();

  const handleOppfolgingError = () => {
    setData(d => ({ ...d, errors: [...d.errors, 'error.baksystemer'], fetching: d.fetching + 1, oppfolgingHasLoaded: true }));
  };

  const incrementFetching = () => {
    setData(d => ({ ...d, fetching: d.fetching + 1 }));
  };

  useEffect(
    () => {
      const handleError = () => {
        incrementFetching();
        setData(d => ({ ...d, errors: [...d.errors, 'error.baksystemer'] }));
      };

      api.fetchBeskjeder()
        .then((r) => {
          incrementFetching();
          dispatch({ type: ADD_BESKJEDER, payload: r });
        }).catch(handleError);

      api.fetchOppgaver()
        .then((r) => {
          setData(d => ({ ...d, oppgaver: r, fetching: d.fetching + 1 }));
        }).catch(handleError);

      api.fetchInnbokser()
        .then((r) => {
          setData(d => ({ ...d, innbokser: r, fetching: d.fetching + 1 }));
        }).catch(handleError);

      /* Skrur midlertidig av denne for å se om det har en effekt på cpu-bruk i database.
      api.fetchAntallBrukernotifikasjoner()
        .then((r) => {
          setData(d => ({ ...d, antallBrukernotifikasjoner: r, fetching: d.fetching + 1 }));
        }).catch(handleError);
       */

      api.fetchInnlogging()
        .then((r) => {
          setData(d => ({ ...d, innlogging: r, fetching: d.fetching + 1 }));
        }).catch(handleError);

      api.fetchOppfolging()
        .then((r) => {
          setData(d => ({ ...d, oppfolging: r, oppfolgingHasLoaded: true, fetching: d.fetching + 1 }));
        }).catch(handleOppfolgingError);

      api.fetchMeldekort()
        .then((r) => {
          setData(d => ({ ...d, meldekort: r, fetching: d.fetching + 1 }));
        }).catch(handleError);

      api.fetchPersonNavn()
        .then((r) => {
          setData(d => ({ ...d, person: r, fetching: d.fetching + 1 }));
        }).catch(() => {
          api.fetchPersonIdent()
            .then(r => {
              setData(d => ({ ...d, identifikator: r, errors: [...d.errors, 'error.baksystemer'], fetching: d.fetching + 1 }));
            })
            .catch(handleError);
        });

      api.fetchSaker()
        .then((r) => {
          const { feilendeBaksystem } = r;
          if (feilendeBaksystem.length > 0) {
            setData(d => ({ ...d, paabegynteSoknader: r, fetching: d.fetching + 1, errors: [...d.errors, 'error.baksystemer'] }));
          } else {
            setData(d => ({ ...d, paabegynteSoknader: r, fetching: d.fetching + 1 }));
          }
        }).catch(handleError);

      api.fetchMeldinger()
        .then((r) => {
          setData(d => ({ ...d, mininnboks: r, fetching: d.fetching + 1 }));
        }).catch(handleError);

      api.fetchSakstema()
        .then((r) => {
          setData(d => ({ ...d, sakstema: r, fetching: d.fetching + 1 }));
        }).catch(handleError);
    }, [],
  );

  const uniqueErrors = data.errors.filter((item, i, ar) => ar.indexOf(item) === i);
  const loading = data.fetching < 10;

  return (
    <PageFrame uniqueErrors={uniqueErrors}>
      <Home data={data} loading={loading} />
    </PageFrame>
  );
};

RenderHome.propTypes = {
  api: ApiType.isRequired,
};

export default RenderHome;
