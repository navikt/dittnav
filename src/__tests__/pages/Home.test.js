import React from 'react';
import wrapIntl from 'js/IntlTestHelper';
import Home from '../../js/pages/Home';
const ReactTestRenderer = require('react-test-renderer');

it('render Home page without props', () => {
  const oppfolging = {};
  const meldekort = {};
  const person = {};
  const identifikator = {};

  const paabegynteSoknader = {
    url: 'https://tjenester-t6.nav.no/',
    antallPaabegynte: 2,
    feilendeBaksystem: [],
  };

  const mininnboks = [];
  const sakstema = {
    sakstemaList: [],
    antallSakstema: 0,
  };

  const fetching = 4;

  const component = ReactTestRenderer
    .create(wrapIntl(<Home oppfolging={oppfolging} meldekort={meldekort} person={person} identifikator={identifikator} paabegynteSoknader={paabegynteSoknader} mininnboks={mininnboks} fetching={fetching} sakstema={sakstema} />));

  expect(component.toJSON()).toMatchSnapshot();
});
