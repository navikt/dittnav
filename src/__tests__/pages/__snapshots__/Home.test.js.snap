// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Home page without props 1`] = `
<div
  className="row"
>
  <div
    className="maincontent side-innhold"
  >
    <div
      className="col-md-12"
      id="dittnav-main-container"
    >
      <section
        className="infomeldinger-list"
      >
        <h1
          className="vekk"
        >
          <span>
            Varsler
          </span>
        </h1>
        <a
          className="message clickable"
          data-ga="Dittnav/Varsel/Paabegynt soknad"
          href="https://tjenester-t6.nav.no/"
        >
          <span
            aria-label="dokument-ikon"
            className="icon document-icon"
          />
          <div
            className="texts"
          >
            <p>
              <span>
                Du har to søknader som du ikke har sendt
              </span>
            </p>
            <p
              id="paabegynte-tekst"
            >
              <span>
                Fortsett innsending
              </span>
            </p>
          </div>
        </a>
      </section>
      <div
        className="dittnav-lenkepanel-top-container"
      >
        <div
          className="dittnav-lenkepanel-liten"
          id={null}
        >
          <a
            alt="Dine saker"
            className="lenkepanel dittnav-lenkepanel-liten-item lenkepanel--border"
            href="http://localhost:9222/saksoversikt/"
          >
            <span
              className="typo-undertittel lenkepanel__heading"
            >
              <span>
                Dine saker
              </span>
            </span>
            <span
              className="lenkepanel__indikator"
            />
          </a>
          <a
            alt="Utbetalinger"
            className="lenkepanel dittnav-lenkepanel-liten-item lenkepanel--border"
            href="http://localhost:9222/utbetalingsoversikt/"
          >
            <span
              className="typo-undertittel lenkepanel__heading"
            >
              <span>
                Dine utbetalinger
              </span>
            </span>
            <span
              className="lenkepanel__indikator"
            />
          </a>
          <a
            alt="Innboks"
            className="lenkepanel dittnav-lenkepanel-liten-item last lenkepanel--border"
            href="http://localhost:9222/mininnboks/"
          >
            <span
              className="typo-undertittel lenkepanel__heading"
            >
              <span>
                Innboks
              </span>
            </span>
            <span
              className="lenkepanel__indikator"
            />
          </a>
        </div>
      </div>
      <h2
        className="typo-undertittel relatert-informasjon__subheader"
      >
        <span>
          Relatert informasjon
        </span>
      </h2>
      <div
        className="relatert-informasjon"
      >
        <nav
          className="relatert-informasjon__links"
        >
          <div
            className="relatert-informasjon__link-container"
          >
            <a
              className="lenke relatert-informasjon__link"
              data-ga="Dittnav/Lenkeliste"
              href="https://arbeidsplassen.nav.no/stillinger"
            >
              Ledige stillinger
            </a>
          </div>
          <div
            className="relatert-informasjon__link-container"
          >
            <a
              className="lenke relatert-informasjon__link"
              data-ga="Dittnav/Lenkeliste"
              href="http://localhost:9222/pselv/publisering/uforetrygd.jsf?context=ut"
            >
              Uføretrygd
            </a>
          </div>
          <div
            className="relatert-informasjon__link-container"
          >
            <a
              className="lenke relatert-informasjon__link"
              data-ga="Dittnav/Lenkeliste"
              href="https://foreldrepenger.nav.no"
            >
              Dine foreldrepenger
            </a>
          </div>
          <div
            className="relatert-informasjon__link-container"
          >
            <a
              className="lenke relatert-informasjon__link"
              data-ga="Dittnav/Lenkeliste"
              href="http://localhost:9222/aktivitetsplan/"
            >
              Aktivitetsplan
            </a>
          </div>
          <div
            className="relatert-informasjon__link-container"
          >
            <a
              className="lenke relatert-informasjon__link"
              data-ga="Dittnav/Lenkeliste"
              href="http://localhost:9222/veiledearbeidssoker/mistet-jobben/registrering"
            >
              Registrer deg som arbeidssøker
            </a>
          </div>
          <div
            className="relatert-informasjon__link-container"
          >
            <a
              className="lenke relatert-informasjon__link"
              data-ga="Dittnav/Lenkeliste"
              href="https://stillingsok.nav.no/pam-stillingsok/lagrede-sok"
            >
              Dine stillingssøk
            </a>
          </div>
          <div
            className="relatert-informasjon__link-container"
          >
            <a
              className="lenke relatert-informasjon__link"
              data-ga="Dittnav/Lenkeliste"
              href="http://localhost:9222/person/personopplysninger"
            >
              Personopplysninger
            </a>
          </div>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
